patch originally taken from "github.com/aste4". i replaced 
their -- and | characters with tmux' default characters and
replaced the corners with empty spaces instead.

diff -uraN a/options-table.c b/options-table.c
--- a/options-table.c	2019-12-01 11:47:31.000000000 +0300
+++ b/options-table.c	2019-12-15 09:30:39.286868246 +0300
@@ -242,6 +242,12 @@
 	  .default_num = ALERT_OTHER
 	},
 
+	{ .name = "alternate-border",
+	  .type = OPTIONS_TABLE_FLAG,
+	  .scope = OPTIONS_TABLE_SESSION,
+	  .default_num = 0
+	},
+
 	{ .name = "assume-paste-time",
 	  .type = OPTIONS_TABLE_NUMBER,
 	  .scope = OPTIONS_TABLE_SESSION,
diff -uraN a/screen-redraw.c b/screen-redraw.c
--- a/screen-redraw.c	2019-07-29 12:50:43.000000000 +0300
+++ b/screen-redraw.c	2019-12-15 09:30:39.287868246 +0300
@@ -501,7 +501,72 @@
 		tty_cursor(tty, i, ctx->statuslines + j);
 	else
 		tty_cursor(tty, i, j);
-	tty_putc(tty, CELL_BORDERS[type]);
+
+    flag = options_get_number(s->options, "alternate-border");
+    if (flag)
+    {
+        switch (type)
+        {
+            case CELL_INSIDE:
+                tty_putc(tty, CELL_BORDERS[type]);
+                break;
+
+            case CELL_OUTSIDE:
+                tty_putc(tty, CELL_BORDERS[type]);
+                break;
+
+            case CELL_LEFTRIGHT:                     /*    |    */
+                tty_putn(tty, "\x7c", 1, 1);         /*    |    */
+                break;                               /*    |    */
+
+            case CELL_TOPBOTTOM:                     /*         */
+                tty_putn(tty, "\x2d", 1, 1);         /* ------- */
+                break;                               /*         */
+
+            case CELL_TOPLEFT:                       /*    +--  */
+                tty_putn(tty, "\x2b", 1, 1);         /*    |    */
+                break;                               /*         */
+
+            case CELL_TOPRIGHT:                      /*  --+    */
+                tty_putn(tty, "\x2b", 1, 1);         /*    |    */
+                break;                               /*         */
+
+            case CELL_BOTTOMRIGHT:                   /*    |    */
+                tty_putn(tty, "\x2b", 1, 1);         /*  --+    */
+                break;                               /*         */
+
+            case CELL_BOTTOMLEFT:                    /*    |    */
+                tty_putn(tty, "\x2b", 1, 1);         /*    +--  */
+                break;                               /*         */
+
+            case CELL_TOPJOIN:                       /*    |    */
+                tty_putn(tty, "\x2b", 1, 1);         /*    |    */
+                break;                               /* ------- */
+
+            case CELL_BOTTOMJOIN:                    /* ------- */
+                tty_putn(tty, "\x2b", 1, 1);         /*    |    */
+                break;                               /*    |    */
+
+            case CELL_LEFTJOIN:                      /*    |    */
+                tty_putn(tty, "\x2b", 1, 1);         /*    |--- */
+                break;                               /*    |    */
+
+            case CELL_RIGHTJOIN:                     /*    |    */
+                tty_putn(tty, "\x2b", 1, 1);         /* ---|    */
+                break;                               /*    |    */
+
+            case CELL_JOIN:                          /*    |    */
+                tty_putn(tty, "\x2b", 1, 1);         /*  --+--  */
+                break;                               /*    |    */
+
+            default:
+                break;
+        }
+    }
+    else
+    {
+        tty_putc(tty, CELL_BORDERS[type]);
+    }
 }
 
 /* Draw the borders. */
